proto-public/pbdns/dns.proto

```diff
@@ -0,0 +1,28 @@
+syntax = "proto3";
+
+package hashicorp.consul.dns;
+
+option go_package = "github.com/hashicorp/consul/proto-public/pbdns";
+
+service DNSService {
+  // Query sends a DNS request over to Consul server and returns a DNS reply message.
+  rpc Query(QueryRequest) returns (QueryResponse) {}
+}
+
+enum Protocol {
+  PROTOCOL_UNSET_UNSPECIFIED = 0;
+  PROTOCOL_TCP = 1;
+  PROTOCOL_UDP = 2;
```

Comment:
Why do we need the `PROTOCOL_` prefix?

Reply:
Linter :(. proto-lint insisted and I was not happy about changing it.

Reply:
We use `buf lint` for all of this. 

IIRC the rationale is is many languages (Go included) cannot properly namespace enum variants and therefore you need to encoded the prefix into the name to help prevent naming collisions. This is especially true for the UNSET/UNSPECIFIED enum variant as you would likely name it identically in all enums.